crates/schema/src/type_for_generate.rs

```diff
@@ -151,6 +155,11 @@ pub enum AlgebraicTypeDef {
     PlainEnum(PlainEnumTypeDef),
 }
 
+thread_local! {
+    /// Used to efficiently extract refs from a def.
+    static EXTRACT_REFS_BUF: RefCell<HashSet<AlgebraicTypeRef>> = RefCell::new(HashSet::new());
```

Comment:
Very nitpicky, but I tend to prefer to move such static / thread-local vars into the function itself, to make it more obvious it's not used anywhere else.